### DISCUSSION TITLE
A limit higher than total number of records at times doesn't return all rows

### DISCUSSION THREAD

**Discussion by titocosta** [supabase/supabase#16503] [Questions] [answered]
I have 4 records in the below table `ads`. If I set `limit` to 40, only 3 ads will be returned. If I set `limit` to 30, all 4 will be returned.
I had a similar situation testing with only 2 and 3 records in the table. 

It appears that when the limit is divisible by the number of records, the query won't return the last record.

```
async function fetchAds(userId=undefined, limit=30) {
  // load the ads
  console.debug(`[ads/show] Loading ads`);
  const query = supabase
    .from("ads")
    .select()

  const { data, error } = await (userId ? query.eq('userId', userId) : query).limit(limit).order('createdAt', { ascending: false });
  
  if(error) {
    console.error(`[ads/show] Supabase returned error loading ads`, error);
    return null;
  } else {
    console.debug(`[ads/show] Loaded ads`, data);
    return data;
  }
}
```
Quite puzzling, am I doing something wrong, is it a bug?

**Comment by GaryAustin1** [comment]
Are you running in an environment like next.js with caching of url requests?
A number of users have been bit changing tables or RLS and the caching being done does not show the changes.  Changing the URL with filters or limit value would bust the cache, or go back to a cached version potentially.

Any RLS involved?

Also you set up your query separately from execution.  Once you have added filters to it you must reinitialize it again before chaining new filters on to the original object (like your query value above).  Otherwise all the filters from before and added will run.

**Comment by titocosta** [comment]
Yes indeed caching is the problem.

How do I avoid NextJS 13 to cache the URL?
Adding `export const dynamic = 'force-dynamic';` to page.tsx doesn't do the trick.

**Reply by GaryAustin1** [reply]
Sorry, I don't use nextjs.  It has come up a bit in Supabase's Discord, but the search there is not great.

I see one comment there:

"Ugh I think it is the environment. Next.js 13 route handler. I added export const revalidate = 0 and now everything is working as expected. Didn't realize these routes are cached just as aggressively as regular server pages."

Maybe that means something to you.

**Reply by titocosta** [reply, marked as answer]
Yes solution is to do `export const revalidate = 0` in the `route.ts` or `page.tsx` files. Doing that in components like I was doing will not work.
